cli/src/program.rs

```diff
@@ -1925,11 +1971,13 @@ where
 fn do_process_program_write_and_deploy(
     rpc_client: Arc<RpcClient>,
     config: &CliConfig,
-    program_data: &[u8],
+    program_data: &[u8], // can be empty, hence we have program_len
     program_len: usize,
-    programdata_len: usize,
-    minimum_balance: u64,
+    min_rent_exempt_program_balance: u64,
```

Comment:
Can you revert this change? There are some tricky code paths around here when it comes to non-upgradeable deployments, and this creates more footguns.

In the case of a buffer that already exists, they need to pass the program's rent-exemption in `min_rent_exempt_program_data_balance`, but in the case of a new deployment, they would need to pass that value in `min_rent_exempt_program_balance`, which is very mistake-prone.

Reply:
Not sure how this change makes it more of a foot-gun, if anything I was hoping to remove a foot-gun or two with it - by making it explicit what `fn do_process_program_write_and_deploy` actually expects as inputs,

it's not strictly needed anymore (it was for enabling `--sign-only` in `deploy`) but it kinda makes me sad to discover this code is "that rigid", unless it's not expected to change much in the future I'd say it's a sign that refactoring(s) are needed to untangle it a bit.

So, do you want me to revert the parts touching "balances" ? Or something else ?